crates/bevy_app/src/plugin.rs

```diff
@@ -56,19 +89,46 @@ use std::any::Any;
 /// # fn damp_flickering() {}
 /// ````
 pub trait Plugin: Downcast + Any + Send + Sync {
-    /// Configures the [`App`] to which this plugin is added.
-    fn build(&self, app: &mut App);
+    /// Returns required sub apps before finalizing this plugin.
+    fn require_sub_apps(&self) -> Vec<InternedAppLabel> {
+        Vec::new()
+    }
 
-    /// Has the plugin finished its setup? This can be useful for plugins that need something
-    /// asynchronous to happen before they can finish their setup, like the initialization of a renderer.
-    /// Once the plugin is ready, [`finish`](Plugin::finish) should be called.
-    fn ready(&self, _app: &App) -> bool {
+    /// Pre-configures the [`App`] to which this plugin is added. This is usually executed before
```

Comment:
Again, please explain why is this useful in the docs <3 I would also probably swap "usually" to "by convention".

Reply:
Done 😉